external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/DockerKrbJDBCIntegrationSuite.scala

```diff
@@ -178,19 +178,7 @@ abstract class DockerKrbJDBCIntegrationSuite extends DockerJDBCIntegrationSuite
             .option("keytab", keytabFullPath)
             .option("principal", principal)
             .option("refreshKrb5Config", "true")
-            .option("query", "SELECT 1")
-            .load()
-        }
-
-        // Set the authentic krb5.conf but doesn't refresh config
-        // so this assertion is expected to fail.
-        intercept[Exception] {
```

Comment:
DB2IntegrationSuite doesn't throw exception here so I'll remove this assertion.

Reply:
Why does db2 throw no exception in this test? cc: @gaborgsomogyi

Reply:
Anyway, since the other krb tests (e.g., `PostgresKrbIntegrationSuite`) refer this test, I think we cannot remove it simply.

Reply:
If `refreshKrb5Config` is not set to true, whether `krb5.conf`  is read or not when a new connection is established can depend on the implementation for each JDBC driver.
We can remove the assertion because the remaining two assertion are important.
The second assertion ensure that `refreshKrb5Config` enforces to refresh.

Reply:
You mean `DB2KrbIntegrationSuite` instead of `DB2IntegrationSuite`, right?

> DB2IntegrationSuite doesn't throw exception here so I'll remove this assertion.

I also confirmed that `DB2KrbIntegrationSuite` fails
```
[info] - SPARK-35226: JDBCOption should accept refreshKrb5Config parameter *** FAILED *** (1 second, 147 milliseconds)
[info]   Expected exception java.lang.Exception to be thrown, but no exception was thrown (DockerKrbJDBCIntegrationSuite.scala:187)
[info]   org.scalatest.exceptions.TestFailedException:
```

Reply:
Ah, Yes. I mean `DB2KrbIntegrationSuite`.